protocol/communities_messenger_token_permissions_test.go

```diff
@@ -1679,3 +1680,136 @@ func checkRoleBasedOnThePermissionType(permissionType protobuf.CommunityTokenPer
+				Decimals:          uint64(18),
+			},
+		},
+		ChatIds: []string{chat.ID},
+	}
+
+	// make bob satisfy channel criteria
+	s.makeAddressSatisfyTheCriteria(testChainID1, bobAddress, channelPermissionRequest.TokenCriteria[0])
+	defer s.resetMockedBalances() // reset mocked balances, this test in run with different test cases
+
+	response, err := s.owner.CreateCommunityTokenPermission(&channelPermissionRequest)
+	s.Require().NoError(err)
+	s.Require().Len(response.Communities(), 1)
+	s.Require().True(s.owner.communitiesManager.IsChannelEncrypted(community.IDString(), chat.ID))
+
+	// reevalate community member permissions in order get encryption keys
+	community, err = s.owner.communitiesManager.GetByID(community.ID())
+	s.Require().NoError(err)
+
+	waitOnChannelKeyToBeDistributedToBob := s.waitOnKeyDistribution(func(sub *CommunityAndKeyActions) bool {
+		action, ok := sub.keyActions.ChannelKeysActions[chat.CommunityChatID()]
+		if !ok || action.ActionType != communities.EncryptionKeyAdd {
+			return false
+		}
+
+		_, ok = action.Members[common.PubkeyToHex(&s.bob.identity.PublicKey)]
+		return ok
+	})
+
+	err = <-waitOnChannelKeyToBeDistributedToBob
+	s.Require().NoError(err)
+
+	// bob receives community changes
+	// channel members should not be empty,
+	// this info is available only to channel members with encryption key
+	_, err = WaitOnMessengerResponse(
+		s.bob,
+		func(r *MessengerResponse) bool {
+			c, err := s.bob.GetCommunityByID(community.ID())
+			if err != nil {
+				return false
+			}
+			if c == nil {
+				return false
+			}
+			channel := c.Chats()[chat.CommunityChatID()]
+			if channel != nil && len(channel.Members) < 2 {
+				return false
+			}
+
+			return channel.Permissions != nil
+		},
+		"no community that satisfies criteria",
+	)
+	s.Require().NoError(err)
+
+	// owner send message to the channel
+	msg := s.sendChatMessage(s.owner, chat.ID, "hello to encrypted channel")
+
+	// bob can read the message
+	response, err = WaitOnMessengerResponse(
+		s.bob,
+		func(r *MessengerResponse) bool {
+			_, ok := r.messages[msg.ID]
+			return ok
+		},
+		"no messages",
+	)
+	s.Require().NoError(err)
+	s.Require().Len(response.Messages(), 1)
+	s.Require().Equal(msg.Text, response.Messages()[0].Text)
+
+	// Simulate backup creation and handling backup message
+	// As a result, bob sends request to resend encryption keys to the owner
+	clock, _ := s.bob.getLastClockWithRelatedChat()
+
+	community, err = s.owner.communitiesManager.GetByID(community.ID())
+	s.Require().NoError(err)
+
+	backupMessage, err := s.bob.backupCommunity(community, clock)
+	s.Require().NoError(err)
+
+	err = s.bob.HandleBackup(s.bob.buildMessageState(), backupMessage, nil)
+	s.Require().NoError(err)
+
+	// regenerate key for the channel in order to check that owner will send keys
+	// on bob request from `HandleBackup`
+	_, err = s.owner.encryptor.GenerateHashRatchetKey([]byte(community.IDString() + chat.CommunityChatID()))
```

Comment:
Just curious, in reality how often do we regenerate keys? 
I recall the 48H? during standup, is it right?
I was wondering if this call should be somehow automatic because otherwise the client that restored a community could be hanging until the `GenerateHashRatchetKey` is triggered by the owner.